src/tests/simulation/src/cluster.rs

```diff
@@ -496,6 +534,134 @@ impl Cluster {
+                tokio::time::sleep(t).await;
+                tracing::info!("restart {name}");
+                madsim::runtime::Handle::current().restart(name);
+            }
+        }))
+        .await;
+    }
+
+    fn addr_to_endpoint(addr: String) -> Result<Endpoint> {
+        let res = Endpoint::from_shared(addr)
+            .map(|endpoint| endpoint.initial_connection_window_size(MAX_CONNECTION_WINDOW_SIZE));
+        match res {
+            Ok(ept) => Ok(ept),
+            Err(_) => Err(anyhow!("Unable to connect")),
+        }
+    }
+
+    // Copy of the implementation in meta_client
+    pub async fn try_build_rpc_channel(addrs: Vec<String>) -> Result<(Channel, String)> {
+        assert!(addrs.len() > 0, "Unable to connect without addresses");
+
+        let endpoints: Vec<_> = addrs
+            .into_iter()
+            .map(|addr| Self::addr_to_endpoint(addr.clone()).map(|endpoint| (endpoint, addr)))
+            .try_collect()?;
+
+        let endpoints = endpoints.clone();
+
+        for (endpoint, addr) in endpoints {
+            match Self::connect_to_endpoint(endpoint).await {
+                Ok(channel) => {
+                    tracing::info!("Connect to meta server {} successfully", addr);
+                    return Ok((channel, addr));
+                }
+                Err(e) => {
+                    tracing::warn!(
+                        "Failed to connect to meta server {}, trying again: {}",
+                        addr,
+                        e
+                    )
+                }
+            }
+        }
+
+        Err(anyhow!("Failed to connect to meta server"))
+    }
+
+    async fn connect_to_endpoint(endpoint: Endpoint) -> Result<Channel> {
+        let res = endpoint
+            .http2_keep_alive_interval(Duration::from_secs(60)) // Self::ENDPOINT_KEEP_ALIVE_INTERVAL_SEC
+            .keep_alive_timeout(Duration::from_secs(60)) // Self::ENDPOINT_KEEP_ALIVE_TIMEOUT_SEC
+            .connect_timeout(Duration::from_secs(5))
+            .connect()
+            .await;
+        match res {
+            Ok(c) => Ok(c),
+            Err(_) => Err(anyhow!("unable to connect to endpoint")),
+        }
+    }
+
+    // TODO: Does not remove it. Marks it for deletion. Rename function
+    /// TODO: return a result here
+    /// remove node from cluster gracefully by informing meta that node is no longer available.
+    pub async fn remove_rand_compute_node(&self) {
+        // where do I get the  client from?
+        // ClusterServiceClient.DeleteWorkerNode
+
+        // TODO: Unable to remove node: Failed to connect to meta server
+        // cannot use advertising meta addr?
+        // How do we resolve the address?
+        // 192, 168, 1, i as u8
+
+        let meta_addrs = (1..self.config.meta_nodes)
+            .map(|i| format!("meta-{i}:5690"))
+            .collect::<Vec<String>>();
+
+        // Which node?
+        let (channel, _) = Self::try_build_rpc_channel(meta_addrs)
```

Comment:
Trying to connect against the simulated meta node. Am I using the wrong address or do I have to do something MadSim specific to send an RPC request against it?